src/gui/tray/usermodel.cpp

```diff
@@ -1014,15 +1014,15 @@ Q_INVOKABLE void UserModel::openCurrentAccountServer()
     QDesktopServices::openUrl(url);
 }
 
-Q_INVOKABLE void UserModel::switchCurrentUser(const int &id)
+Q_INVOKABLE void UserModel::setCurrentUserId(const int &id)
 {
     if (_currentUserId < 0 || _currentUserId >= _users.size())
         return;
     
     _users[_currentUserId]->setCurrentUser(false);
     _users[id]->setCurrentUser(true);
     _currentUserId = id;
-    emit newUserSelected();
+    emit currentUserChanged();
```

Comment:
this is missing a proper check before doing anything
```
if (_cuurentUserId == id) {
    return;
}
```

would allow to get rid of this check in qml code and would be much more the "proper" way of doing a setter

Reply:
Very much agree, nice catch -- I added this check to the existing check in this method and removed the check in the QML